### DISCUSSION TITLE
shape function and its gradients

### DISCUSSION THREAD

**Discussion by YangZhao16** [thelfer/MFrontGenericInterfaceSupport#91] [Q&A]
Hi @thelfer,

In Mfront, is there a function to call to get the values of the shape functions and their gradients at the Gauss point?

Thanks,
Yang.

**Comment by thelfer** [comment]
@YangZhao16 Basically no.
The computation of the gradients (strain for example) and the assembly of the residual and assembly matrix (if required, it does not make sense in FEM explicit or FFT) are devoted to the calling solver(s).
Of course, there can be workarounds in specific cases and specific solvers..
As usual, it would be better to know what you intent to do, because there may be another approach than requesting those "details" locally
Thomas

**Reply by YangZhao16** [reply]
Hi @thelfer,

I was trying to implement enhanced assumed strain finite element with Mfront to model fracture. One component in my work requires evaluation of the gradient of shape functions at the material point level.

I am using CST element right now, so the gradients should be constant over an element. One way I thought of to get the gradients is to feed in the coordinates of the nodes of the elements to Mfront as material properties, and then calculate the gradients (which will be perpendicular to the edges). I just wonder if there exists more straightforward methods.

Again, thanks for your quick response!

**Reply by thelfer** [reply]
@YangZhao16 Remind me: what solver do you use ?
